### PR TITLE
Delete Tracking Project

### PR DESCRIPTION
**Pull request by Angelk90**
https://user-images.githubusercontent.com/20476002/159308393-3f865c9b-bb1a-4440-8245-5f4539187301.mov

### CODE REVIEW

File: src/stores/ui.store.ts [hunk 1]
```diff
@@ -583,6 +592,24 @@ end tell`)
           break
         }
 
+        // delete key
+        case 51: {
+          switch (store.focusedWidget) {
+            case FocusableWidget.PROJECT_SELECT: {
+              if(store.projects.length === 0) break
```
Comment (ospfranco) [inline src/stores/ui.store.ts:599]: deletion without confirmation is too dangerous, at least this needs a add a confirmation dialog

File: src/stores/ui.store.ts [hunk 2]
```diff
@@ -392,6 +392,14 @@ end tell`)
       }
       store.query = ''
     },
+    deleteTrackingProject: (id: string) => {
+      if (store.currentlyTrackedProjectId === id) {
```
Comment (ospfranco) [inline src/stores/ui.store.ts:396]: isn't this code repeated from the key handler below?

**Review by ospfranco — CHANGES_REQUESTED** [review]

File: src/stores/ui.store.ts [hunk 3]
```diff
@@ -519,6 +527,7 @@ end tell`)
             }
 
             case FocusableWidget.PROJECT_SELECT: {
+              if(store.projects.length === 0) break
```
[the remark this answers is not in the document]
Comment (Angelk90) [reply src/stores/ui.store.ts:530]: What do you mean?

Schermata 2022-03-21 alle 17 41 14

File: src/stores/ui.store.ts (hunk 2 above)
Comment (Angelk90) [reply src/stores/ui.store.ts:396]: I do not understand.
Here check if the project you are trying to delete is the same as the one you are running, first tow it and then remove it.

File: src/stores/ui.store.ts (hunk 1 above)
Comment (Angelk90) [reply src/stores/ui.store.ts:599]: I don't know how to implement a dialog for react-native for macos I tried dialog but it gives problems, then I don't know if you might like it as ui and how to handle the request/confirmation communication between the store and view widget.

I think it's quite complex if you want to use the dialog.

Take the implemented code as an example if you want to implement this possibility in the future.

File: src/stores/ui.store.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ospfranco) [reply src/stores/ui.store.ts:530]: ah, I forgot, did I add that, then it's fine, forget about it

File: src/stores/ui.store.ts (hunk 1 above)
Comment (ospfranco) [reply src/stores/ui.store.ts:599]: sorry, this needs a confirmation dialog, the deletion code is easy to do, but the UX needs to work before I implement it

**Pull request closed without merging**